src/appmixer/servicenow/lib.js

```diff
@@ -0,0 +1,256 @@
+const crypto = require('crypto');
+
+'use strict';
+
+const pathModule = require('path');
+
+function getBasicAuth(username, password) {
+
+    return Buffer.from(username + ':' + password).toString('base64');
+}
+
+function getCacheKey(obj) {
+    // Convert the object to a JSON string
+    const str = JSON.stringify(obj);
+
+    // Create a hash using the SHA-256 algorithm
+    const hash = crypto.createHash('sha256');
+    hash.update(str);
+
+    // Get the resulting hash as a hexadecimal string
+    return hash.digest('hex');
+}
+
+async function callEndpointCached(context, options) {
+
+    const key = getCacheKey({ ...options, token: context.auth.username + context.auth.password });
+
+    const cached = await context.staticCache.get(key);
+    if (cached) {
+        return { data: cached };
+    }
+
+    const { data } = await callEndpoint(context, options);
+
+    await context.staticCache.set(
+        key,
+        data,
+        context.config.listBasesCacheTTL || (2 * 60 * 1000) // 120s
+    );
+
+    return { data };
```

Comment:
I think this block should also be inside a `context.lock(key)`. Imagine a flow where different types of eg. issues (table `incident`) are created. Lets say: errors, questions, feature requests. Each `CreateRecord` component will have the same inspector, but different fields (eg repro steps for error, but not for the other two types).

Without lock there will most likely be 3 calls to `callEndpointCached` even though it is cached - because they will fire at the same time.